subprojects/files/src/main/java/org/gradle/internal/file/DefaultFileHierarchySet.java

```diff
@@ -82,6 +111,11 @@ public String toString() {
             this.rootNode = new Node(path);
         }
 
+        PrefixFileSet(String rootPath) {
```

Comment:
I pushed some changes so the `File` constructor delegates to this constructor and `plus(File)` delegates to `plus(String)`. I think that makes the code easier to understand.

Reply:
👍